File: examples/weather/apple/WeatherApp/LiveBridge.swift
import App
import Foundation
import Shared

/// The generated `CoreBridge`, implemented over BoltFFI's `CoreFfi`: bytes in,
/// bytes out, nothing else. This is the only file in the app that knows the
/// Rust core exists, and it lives in the app target (not WeatherKit) so that
/// SwiftUI previews don't need to load the Rust framework.
///
/// `nonisolated` because the target builds with `MainActor` as its default
/// isolation, and `CoreBridge`'s requirements are not actor-isolated.
/// `@unchecked Sendable` because `CoreFfi` is a class Swift cannot prove
/// `Sendable`; its state is a handle into a mutex-guarded `Bridge` on the Rust
/// side, so calling it from any task is safe.
nonisolated struct LiveBridge: CoreBridge, @unchecked Sendable {
    private let ffi = CoreFfi()

    func update(_ event: [UInt8]) -> [UInt8] {
        [UInt8](ffi.update(data: Data(event)))
    }

    func resolve(_ id: UInt32, _ output: [UInt8]) -> [UInt8] {
        [UInt8](ffi.resolve(id: id, data: Data(output)))
    }

    func view() -> [UInt8] {
        [UInt8](ffi.view())
    }
}

File: examples/weather/Android/app/src/main/java/com/crux/example/weather/core/LiveBridge.kt
package com.crux.example.weather.core

import com.crux.example.weather.CoreBridge
import com.crux.example.weather.CoreFfi
import javax.inject.Inject
import javax.inject.Singleton

/// The generated `CoreBridge`, implemented over BoltFFI's `CoreFfi`: bytes in,
/// bytes out, nothing else. This is the only file in the app that knows the
/// Rust core exists — swap it for a fake and the rest of the shell is
/// unchanged.
@Singleton
class LiveBridge
    @Inject
    constructor() : CoreBridge {
        private val coreFfi = CoreFfi()

        override fun update(event: ByteArray): ByteArray = coreFfi.update(event)

        override fun resolve(
            id: UInt,
            output: ByteArray,
        ): ByteArray = coreFfi.resolve(id, output)

        override fun view(): ByteArray = coreFfi.view()
    }
